creating-pull-requests.md

```diff
@@ -0,0 +1,131 @@
+# Creating a Pull Request
+
```

Comment:
Add a table of contents.

Please note that there is a script in the `tools` folder to generate one. We only need to update it so it also works for this file

Reply:
done!